### ISSUE TITLE
[Truffle] Issue with default block arguments

### ISSUE THREAD

**Issue by bjfish** [jruby/jruby#2878]
Found in MRI test_syntax.rb:
Example
```ruby
def yieldme
  yield
end

yieldme {  | methods = [['map', 'no'], ['inject([])', 'with']] |
    puts "METHODS `#{methods}`"
}
```
Expected (ruby, jruby)
```
METHODS `[["map", "no"], ["inject([])", "with"]]`
```
Actual
```
METHODS `map`
```

**Comment by chrisseaton** [comment]
It's not blocks - it's any default argument that has an array literal on the RHS.

```
def foo(methods=[14])
  p methods
end

foo
```

**Comment by chrisseaton** [comment]
Fixed in 146856d0b9fc3168de23314904b766e10e03069c
